js/chat.js

```diff
@@ -41,34 +39,45 @@ ChatCreator.prototype.systemMessage = function(val){
 	}]);
 };
 
-ChatCreator.prototype.commandParse = function(val){
-	if(!this.commands.has(val))return false;
-
-	switch(val){
-		case '/online':
-			(function(){
+Chat.prototype.setupCommands = function() {
+	this.commands = new Map([
+		[
+			'/online'
+			,function() {
 				var a = '';
-				for(var i of this.users.values()){
+
+				for(var i of this.users.values()) {
 					a += (a===''?'':', ') + (i.idle?'[i] ':'') + i.name;
 				}
 
 				this.systemMessage('online: ' + a);
-			}).apply(this);
-			break;
+			}.bind(this)
+		]
+		,[
+			'/clear'
```

Comment:
Jezeli poprawiales cos to ok, ale u mnie po wpisaniu /clear czyscilo chat
tylko ze jak ktos napisal to znow ladowalo cale okno :)

W dniu sobota, 22 sierpnia 2015 Comandeer <notifications@github.com>
napisał(a):

> In js/chat.js
> <https://github.com/CodersCommunity/czatrepajr/pull/18#discussion_r37695184>
> :
>
> >  					a += (a===''?'':', ') + (i.idle?'[i] ':'') + i.name;
> >  				}
> >
> >  				this.systemMessage('online: ' + a);
> > -			}).apply(this);
> > -			break;
> > +			}.bind(this)
> > +		]
> > +		,[
> > +			'/clear'
>
> Jak na moje oko to działa, hmm…
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CodersCommunity/czatrepajr/pull/18/files#r37695184>.
>